dev/devicelab/bin/tasks/flutter_gallery_macos__start_up.dart

```diff
@@ -2,11 +2,11 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:flutter_devicelab/framework/devices.dart';
 import 'package:flutter_devicelab/framework/framework.dart';
 import 'package:flutter_devicelab/tasks/perf_tests.dart';
 
 Future<void> main() async {
-  // It's intended to use the Gallery startup test as a smoke test on macOS
-  // Catalina.
+  deviceOperatingSystem = DeviceOperatingSystem.macos;
```

Comment:
rather than renaming the old file, don't you want to copy it to create a new one?

Reply:
I removed the file in fd94568937c57b364172a4bc57e223260be1b8eb, created a new one in 3764fdbe3653b1b71c58823f9ed87b91a290efa8, and copied it over in 27be22991ab7e252d2b4397f7f32856a0011ed0c. The diff says otherwise.

Reply:
I could be misunderstanding the suggestion. Is the suggestion that I keep both tests/tasks?

Reply:
Don't we want to test starting on android and macOS?

Reply:
Yes, you're totally right and I missed that. Thanks!

Reply:
@christopherfujino I'll open another PR to do the renaming for the android test and then try to land this one. How does that sound?

Reply:
As a side note, is this a benchmark, or just pass/fail test? If the latter, can we track eventually moving this out of the devicelab?

Reply:
Looks like this is the bug; I don't see this particular task listed on it though: https://github.com/flutter/flutter/issues/65790

Reply:
> @christopherfujino I'll open another PR to do the renaming for the android test and then try to land this one. How does that sound?

You'll have to check with @keyonghan about if renaming the android test is the right thing to do, it might mess up benchmark tracking...unless it's not a benchmark?

Reply:
However if this PR simply adds a new test, that SGTM

Reply:
Again, if this is not a benchmark I think it should be with a TODO to migrate it, linked to #65790 though

Reply:
@christopherfujino this is a benchmark, it measures the amount of time until the first frame is shown.

Reply:
cc @CaseyHillers

Reply:
Renaming `flutter_gallery_mac__start_up` to `flutter_gallery_macos__start_up` will cause historical benchmark data disconnected. /cc @zanderso

Reply:
Just to clarify, `flutter_gallery_mac__start_up` was running on a `Mac_android` host. I'm not certain, but I believe the benchmark was measuring the time to first frame for `flutter_gallery` running on an Android.

I want to bring up new target that measures the same thing for macOS, called `flutter_gallery_macos__start_up`.

It would be confusing if we left in both
`flutter_gallery_mac__start_up`
`flutter_gallery_macos__start_up`

I propose we rename `flutter_gallery_mac__start_up` to `flutter_gallery_android__start_up` in a separate PR and land that before this.

Then we land this PR to bring up `flutter_gallery_macos__start_up`.

Reply:
> Just to clarify, `flutter_gallery_mac__start_up` was running on a `Mac_android` host. I'm not certain, but I believe the benchmark was measuring the time to first frame for `flutter_gallery` running on an Android.
> 
> I want to bring up new target that measures the same thing for macOS, called `flutter_gallery_macos__start_up`.
> 
> It would be confusing if we left in both `flutter_gallery_mac__start_up` `flutter_gallery_macos__start_up`
> 
> I propose we rename `flutter_gallery_mac__start_up` to `flutter_gallery_android__start_up` in a separate PR and land that before this.
> 
> Then we land this PR to bring up `flutter_gallery_macos__start_up`.

To further clarify, Keyong is saying that this diff will affect our benchmark tracking data

![image](https://user-images.githubusercontent.com/7856010/187254509-b86c7b4d-52ef-4a56-bb84-103466e3c813.png)

Reply:
Got it 💯 . I'd like to forward the decision to someone on the infra since they will know more.

Reply:
You can change the file name and `task_name` but leave the `name` as `flutter_gallery_mac__start_up`, that will prevent the benchmark breakage and will leave a nugget that that data is for Android.

Reply:
Great idea. I wish I thought of that sooner!

Reply:
SKIA has been using `task_name` instead of builder target `name` to collect benchmarks. So changing the `task_name` will still cause breakage.

Reply:
Oh, sorry I misunderstood @keyonghan. So we can change `name` and it's safe? Or are both locked in?

The mapping of skia data to these names is a continued painpoint.  I wish we could switch all these skia benchmarks to use a guid identifier or something similar, and then hook up display names and paths to the target files as something separate...

Reply:
Changing `name` will not affect benchmark metrics. (note that this will affect the columns in go/flutter-build.)
IIRC, we changed the SKIA data to be affiliated with `task_name` instead of `name` to be independent of `name` changes, for example a task migration from a Mac platform to Linux.

Reply:
What if we: 
1. Rename `flutter_gallery_mac__start_up` to `flutter_gallery_android__start_up`.
2. Create a new symlinked file called `flutter_gallery_mac__start_up`, points to `flutter_gallery_android__start_up`.
3. Leave the current target as is.
4. Bring up a new target called `Mac_android flutter_gallery_android__start_up` with `task: flutter_gallery_android__start_up`

Reply:
I don't think it's a good idea to keep two duplicate tests running the same thing, especially considering our limited devicelab bot resource.

To make things easier, if this is just a one-time change for a single test that will make the world more clear, you can work with the engine gardener to make sure there is no regression happening during the name change period. The regression logic is based on recent 7 commits data, whenever there is no performance regression in one day, we should be good. But we need to get an approval from the test owner whether they are okay with the data disconnection.

Reply:
If there are other tests/benchmarks that we can launch an Android app from a mac host, then we can just delete `flutter_gallery_mac__start_up` since the benchmark numbers are the same as from a Linux host: https://flutter-flutter-perf.skia.org/e/?begin=1621542543&end=1661881565&queries=test%3Dflutter_gallery__start_up%26test%3Dflutter_gallery_mac__start_up&requestType=0

@keyonghan Do we have other Android device / macOS host tests and benchmarks?

Reply:
> If there are other tests/benchmarks that we can launch an Android app from a mac host, then we can just delete `flutter_gallery_mac__start_up` since the benchmark numbers are the same as from a Linux host: https://flutter-flutter-perf.skia.org/e/?begin=1621542543&end=1661881565&queries=test%3Dflutter_gallery__start_up%26test%3Dflutter_gallery_mac__start_up&requestType=0
> 
> @keyonghan Do we have other Android device / macOS host tests and benchmarks?

This is a good point; I think there are two behaviors we are interested in that are getting conflated here: 1) benchmark of time it takes the Flutter gallery on android to start up; and 2) whether or not building and starting the gallery on different Flutter host/target platform combinations works.

For 1, it's probably enough to just benchmark Linux/Android. For 2, these tests don't need to be devicelab tests, and can/should be added to test.dart _runBuildTests() https://github.com/flutter/flutter/blob/master/dev/bots/test.dart. It looks like we already build flutter_gallery (which I'm guessing is the old gallery).

Reply:
> @keyonghan Do we have other Android device / macOS host tests and benchmarks?

Currently we have three more mac/android benchmarks:
- Mac_android microbenchmarks
- Mac_android hot_mode_dev_cycle__benchmark
- Mac_android hello_world_android__compile

Reply:
Good point @zanderso.  I tried to audit all the Mac/Android benchmarks in https://github.com/flutter/flutter/issues/74522 but that one was special per https://github.com/flutter/flutter/issues/73907 (#94980) because this was originally used to test the tool on different macOS versions? #38567

Anyway you're right, we should just delete `flutter_gallery_mac__start_up`.

Reply:
> * Mac_android microbenchmarks

We kept this on Mac because it was flaky on Linux, though maybe that isn't true anymore: https://github.com/flutter/flutter/issues/74522#issuecomment-1009494212 https://github.com/flutter/flutter/issues/95366

> * Mac_android hot_mode_dev_cycle__benchmark
> * Mac_android hello_world_android__compile

These should stay on `Mac_android` as they are testing performance/behavior between the Mac and Android devices.  Per https://github.com/flutter/flutter/issues/74522

> #### Host-dependent tests (should not be moved)
> * run_release_test
> * hello_world_android__compile
> * hot_mode_dev_cycle__benchmark

Reply:
This PR is updated with the changes discussed in the thread. The diff does not accurately reflect my intentions so to be clear:

I created the new task (`flutter_gallery_macos__start_up`) in 6be515f6e1d91e65a6b5aa149da7ae904d3a2e78 and deleted `flutter_gallery_mac__start_up` in ce8ee07258a646ecec099719584ed74c5e115299 

The diff, on the other hand, shows that I simply renamed the file.

Reply:
> It's probably enough to just benchmark Linux/Android.

It doesn't look like we have a benchmark that records time to first frame on android. 

`dev/devicelab/bin/tasks/`
```
flutter_gallery_android__compile
/* MISSING: flutter_gallery_android__start_up */
flutter_gallery_ios__compile
flutter_gallery_ios__start_up
flutter_gallery_macos__compile
flutter_gallery_macos__start_up
flutter_gallery_win_desktop__compile
flutter_gallery_win_desktop__start_up
```

Reply:
> > @christopherfujino It's probably enough to just benchmark Linux/Android.
> 
> It doesn't look like we have a benchmark that records time to first frame on android.
> 
> `dev/devicelab/bin/tasks/`
> 
> ```
> flutter_gallery_android__compile
> /* MISSING: flutter_gallery_android__start_up */
> flutter_gallery_ios__compile
> flutter_gallery_ios__start_up
> flutter_gallery_macos__compile
> flutter_gallery_macos__start_up
> flutter_gallery_win_desktop__compile
> flutter_gallery_win_desktop__start_up
> ```

unforunately android is usually implicit in these old devicelab tests

https://github.com/flutter/flutter/blob/master/dev/devicelab/bin/tasks/flutter_gallery__start_up.dart